### PR TITLE
addition of flux area correction factors

### PR DESCRIPTION
**Pull request by mvertens**
This PR adds area flux correction factors as is done in CPL7. Remove some debug writes of coupler fields. And remove state_getfldptr as it was unused.

Also sets the number of OpenMP threads according to the coupler namelist rather than env variables.

### CODE REVIEW

File: src/cpl/nuopc/rof_import_export.F90 [hunk 1]
```diff
@@ -489,184 +530,99 @@ end subroutine SetScalarField
...
 
+    use ESMF, only : ESMF_StateGet, ESMF_FieldGet, ESMF_Field
+
     ! input/output variables
     type(ESMF_State)    , intent(in)    :: state
     character(len=*)    , intent(in)    :: fldname
     integer             , intent(in)    :: begr
     integer             , intent(in)    :: endr
     real(r8)            , intent(in)    :: area(begr:endr)
+    real(r8), optional  , intent(in)    :: areacor(:)
```
Comment (billsacks) [inline src/cpl/nuopc/rof_import_export.F90:547]: As in https://github.com/ESCOMP/RTM/pull/24#discussion_r593500320 it seems problematic to have an optional argument appear before required arguments, here and in state_setexport. And as in https://github.com/ESCOMP/RTM/pull/24#discussion_r593500320, my preferred solution would be to have state_getimport directly use the module variable med2mod_areacor and state_setexport directly use mod2med_areacor.

**Review by billsacks — CHANGES_REQUESTED** [review]
Same comment/request as in https://github.com/ESCOMP/RTM/pull/24

**Review by ekluzek — APPROVED** [review]
I agree with @billsacks objection. areacor should be a required argument or used as a module variable. But, other than that it looks fine. Note, this includes some changes to OpenMP threads so we should ensure that's included in the testing (which it will for the CTSM tests).

**Comment by billsacks** [comment]
Thanks @ekluzek . Note that the changes to threading are only in the nuopc cap, and threading with nuopc currently is *not* covered in our aux testing.

**Comment by billsacks** [comment]
I discussed this with @mvertens . She pointed out that you don't always want to do the area correction – such as for states. We came up with an interface to these routines that I like better to handle this:
- Don't pass the area correction factors; instead, use the module variables (this way you don't accidentally mix up which area correction factor to use for the import/export)
- Add a new, _required_ argument: `do_area_correction`; the area correction will be applied if this is true. By making it required, we force someone adding a new field to consider whether it should be true or false for their new field.

**Review by billsacks — APPROVED** [review]

**Comment by billsacks** [comment]
I have confirmed that my refactor did *not* change answers for the test `SMS_D_Ld5_Vnuopc.f10_f10_mg37.I2000Clm50BgcCrop.cheyenne_intel.mosart-default`, which I ran in the context of CESM's nuopc_dev branch. I will merge this shortly.

**Pull request merged**